File: tests/parsers/samples/code/f90/simple.f90
module simple_metrics
  use complex_metrics
  implicit none
contains
  subroutine summarize(baseline, rms)
    real, intent(out) :: baseline, rms
    real :: values(3)
    values = (/ 1.0, 2.0, 3.0 /)
    baseline = distance(10, 3)
    rms = root_mean_square(values, 3)
  end subroutine summarize
end module simple_metrics

File: tests/parsers/samples/code/ex/simple.ex
defmodule SimpleMetrics do
  alias ComplexMetrics, as: Complex

  def run_simple do
    analyzer = Complex.build_default()
    baseline = Complex.distance(10, 3, analyzer.scale)
    rms = Complex.root_mean_square([1.0, 2.0, 3.0])
    %{baseline: baseline, rms: rms}
  end

  def summarize do
    metrics = run_simple()
    "baseline=#{Float.round(metrics.baseline, 2)}, rms=#{Float.round(metrics.rms, 2)}"
  end
end
